inspirehep/migrator/tasks.py

```diff
@@ -198,33 +195,101 @@ def continuous_migration():
             while r.llen("legacy_records"):
                 raw_record = r.lrange("legacy_records", 0, 0)
                 if raw_record:
-                    migrate_and_insert_record(zlib.decompress(raw_record[0]))
+                    insert_records_for_migration(zlib.decompress(raw_record[0]))
                     db.session.commit()
                 r.lpop("legacy_records")
         finally:
+            migrate_from_mirror.ignore_resuls = False
+            migration_tasks = migrate_from_mirror(disable_orcid_push=False)
+            if migration_tasks:
+                with allow_join_result():
+                    if force_wait:
+                        migration_tasks.wait()
+                    else:
+                        migration_tasks.wait(timeout=max_timeout)
+                    migration_tasks.forget()
             lock.release()
     else:
         LOGGER.info("Continuous_migration already executed. Skipping.")
 
 
 @shared_task(ignore_result=False, queue="migrator")
-@disable_orcid_push
-def migrate_recids_from_mirror(prod_recids):
+def create_records_from_mirror_recids(recids):
+    """Task which migrates records
+    Args:
+        recids: record to migrate
+    Returns:
+         set: set of properly processed records uuids
+    """
     models_committed.disconnect(index_after_commit)
-    index_queue = []
-    for recid in prod_recids:
-        with db.session.begin_nested():
-            record = migrate_record_from_mirror(LegacyRecordsMirror.query.get(recid))
-            if record and not record.get("deleted"):
-                index_queue.append(
-                    InspireRecordIndexer()._process_bulk_record_for_index(record)
+    processed_records = set()
+    for recid in recids:
+        try:
+            with db.session.begin_nested():
+                record = migrate_record_from_mirror(
+                    LegacyRecordsMirror.query.get(recid)
                 )
+        except Exception as e:
+            logger.error(f"Cannot process record {recid}: {e}")
+            continue
+        if record and not record.get("deleted", False):
```

Comment:
I think this is wrong, as a the next steps need to happen for deleted records too.